### PR TITLE
Bug/fix nft gesture experience

### PR DESCRIPTION
**Pull request by Cal-L**
**Description**

This PR is a rework of the modal that wraps CollectibleModal.js. Few things that changed:
- Created reusable full screen modal that wraps CollectibleModal.js
- Added redash library, which has helper methods for animation code
- Moved the collectible modal to MainNavigator
- Introduce Typescript to es-lint
- Update other files to confirm with reusable modal interface

**TODO: Tests**
- Test account with NFT on iOS
- Test with video

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Any added code is fully documented

**Issue**
https://github.com/MetaMask/metamask-mobile/pull/2731#pullrequestreview-692300582

### CODE REVIEW

**Review by ibrahimtaveras00 — APPROVED** [review]
Fix looks good on both OS's, QA Passed

File: app/components/UI/CollectibleModal/reusable-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+		</View>
+	);
+});
+
+const styles = StyleSheet.create({
+	container: {
+		flex: 1
+	},
+	overlayBackground: {
+		backgroundColor: 'rgba(0,0,0,0.7)',
```
Comment (sethkfman) [inline app/components/UI/CollectibleModal/reusable-modal.tsx:209]: Could you move this into the colors.xml?

File: .eslintrc.js [hunk 2]
```diff
@@ -20,6 +20,11 @@ module.exports = {
 		waitFor: true,
 		__DEV__: true
 	},
+	settings: {
+		'import/resolver': {
+			typescript: {} // this loads <rootdir>/tsconfig.json to eslint
```
Comment (sethkfman) [inline .eslintrc.js:25]: Check out this repo to standardize our linter rules for typescript: https://github.com/MetaMask/eslint-config/tree/main/packages/typescript

Comment (Cal-L) [reply .eslintrc.js:25]: @sethkfman My bad didn't see this. Will address separately

**Pull request merged**